<!DOCTYPE html>
  <head>

 <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="">
<meta name="keywords" content=" ">
<title>DEG analysis with `limma`  | Manuals</title>
<link rel="stylesheet" type="text/css" href="../css/syntax.css">
<link rel="stylesheet" type="text/css" href="../css/font-awesome.min.css">
<!--<link rel="stylesheet" type="text/css" href="../css/bootstrap.min.css">-->
<link rel="stylesheet" type="text/css" href="../css/modern-business.css">
<link rel="stylesheet" type="text/css" href="../css/lavish-bootstrap.css">
<link rel="stylesheet" type="text/css" href="../css/customstyles.css">
<link rel="stylesheet" type="text/css" href="../css/theme-blue.css">
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-cookie/1.4.1/jquery.cookie.min.js"></script>
<script src="../js/jquery.navgoco.min.js"></script>
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/js/bootstrap.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/anchor-js/2.0.0/anchor.min.js"></script>
<script src="../js/toc.js"></script>
<script src="../js/customscripts.js"></script>
<script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
<link rel="shortcut icon" href="../common_images/favicon.ico" type="image/x-icon">
<!-- HTML5 Shim and Respond.js IE8 support of HTML5 elements and media queries -->
<!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
<script src="https://oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
<![endif]-->







 


<script>
  $(function () {
      $('[data-toggle="tooltip"]').tooltip()
  })
</script>



  </head>

<body>

   <!-- Navigation -->
<nav class="navbar navbar-inverse navbar-fixed-top" role="navigation">
    <div class="container topnavlinks">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <a class="fa fa-home fa-lg navbar-brand" href="home.html">&nbsp;<span class="projectTitle"> Longevity Tools</span></a>

        </div>
        <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
            <ul class="nav navbar-nav navbar-right">
                <!-- entries without drop-downs appear here -->
                <!-- conditional logic to control which topnav appears for the audience defined in the configuration file.-->
                









                
                
                
                
                <li><a href="http://www.longevitygenomics.org/" target="_blank">Longevity Genomics</a></li>
                
                
                
                
                
                <li><a href="https://github.com/tgirke/longevityTools" target="_blank">Github Repo</a></li>
                
                
                
                


                <!-- entries with drop-downs appear here -->
                <!-- conditional logic to control which topnav appears for the audience defined in the configuration file.-->

                <li class="dropdown">
                    
                    
                    
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Resources<b class="caret"></b></a>
                    <ul class="dropdown-menu">
                        
                        
                        
                        <li><a href="https://cran.r-project.org/" target="_blank">CRAN</a></li>
                        
                        
                        
                        
                        
                        <li><a href="http://bioconductor.org/" target="_blank">Bioconductor</a></li>
                        
                        
                        
                        
                        
                        <li><a href="https://guides.github.com/activities/hello-world/" target="_blank">GitHub</a></li>
                        
                        
                        
                        
                        
                        <li><a href="http://jekyllrb.com/" target="_blank">Jekyll</a></li>
                        
                        
                        
                        

                    </ul>
                </li>
                
                


                <!-- special insertion -->

               
                <!-- Send feedback function -->
<script>
function SendLinkByMail(href) {
var subject= "Manuals feedback";
var body = "I have some feedback about the DEG analysis with `limma` page: ";
body += window.location.href;
body += "";
var uri = "mailto:?subject=";
uri += encodeURIComponent(subject);
uri += "&body=";
uri += encodeURIComponent(body);
window.location.href = uri;
}
</script>

<li><a href="javascript:(function()%7BSendLinkByMail()%3B%7D)()%3B" target="_blank"><i class="fa fa-envelope-o"></i> Feedback</a></li>


                <!--uncomment this block if you want simple search instead of algolia-->
                <li>
                     <!--start search-->
                    <div id="search-demo-container">
                        <input type="text" id="search-input" placeholder="search...">
                        <ul id="results-container"></ul>
                    </div>
                    <script src="../js/jekyll-search.js" type="text/javascript"></script>
                    <script type="text/javascript">
                        SimpleJekyllSearch.init({
                            searchInput: document.getElementById('search-input'),
                            resultsContainer: document.getElementById('results-container'),
                            dataSource: '../search.json',
                            searchResultTemplate: '<li><a href="{url}" title="DEG analysis with `limma`">{title}</a></li>',
                        noResultsText: 'No results found.',
                                limit: 10,
                                fuzzy: true,
                        })
                    </script>
                     <!--end search-->
                </li>

            
        </div>
        <!-- /.container -->
</nav>


    <!-- Page Content -->
    <div class="container">
        <div class="col-lg-12">&nbsp;</div>


<!-- Content Row -->
<div class="row">
    <!-- Sidebar Column -->
    <div class="col-md-3">

        <script>

            $(document).ready(function() {
                // Initialize navgoco with default options
                $("#mysidebar").navgoco({
                    caretHtml: '',
                    accordion: true,
                    openClass: 'active', // open
                    save: false, // leave false or nav highlighting doesn't work right
                    cookie: {
                        name: 'navgoco',
                        expires: false,
                        path: '/'
                    },
                    slide: {
                        duration: 400,
                        easing: 'swing'
                    }
                });

                $("#collapseAll").click(function(e) {
                    e.preventDefault();
                    $("#mysidebar").navgoco('toggle', false);
                });

                $("#expandAll").click(function(e) {
                    e.preventDefault();
                    $("#mysidebar").navgoco('toggle', true);
                });

            });

        </script>


        









        <ul id="mysidebar" class="nav">

            <span class="siteTagline"><img src="../common_images/logo.png"></span>
            <span class="versionTagline"></span>

            
            
            
                
            
            <li><a href="#">Overview</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/home.html">Introduction</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_current_projects.html">Current Projects</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_getting_started.html">Getting Started</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Downloads</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_downloads.html">Download of Analysis Results</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_software.html">Software</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">longevityToolsOverview</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_1.html">1. Introduction</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_2.html">2. Getting Started</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_3.html">3. Overview of object clases, methods and functions</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_4.html">4. Workflow vignettes</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_5.html">5. Funding</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_6.html">6. Version information</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_7.html">7. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">longevityTools_eQTL</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eQTL_01.html">1. Introduction</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eQTL_02.html">2. General Analytical Approach</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eQTL_03.html">3. Getting Started</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eQTL_04.html">4. Setup of working environment</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eQTL_05.html">5. Sample data from GTEx</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eQTL_06.html">6. Utilities</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eQTL_07.html">7. SNP clumping using PLINK</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eQTL_08.html">8. Funding</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eQTL_09.html">9. Version information</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eQTL_10.html">10. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">longevityTools</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eDRUG_01.html">1. Introduction</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eDRUG_02.html">2. Getting Started</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eDRUG_03.html">3. Setup of working environment</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eDRUG_04.html">4. Data downloads</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eDRUG_05.html">5. Pre-processing of CEL files</a></li>
                    

                    

                    
                    
                    
                    
                    <li class="active"><a href="../mydoc/mydoc_longevityTools_eDRUG_06.html">6. DEG analysis with `limma`</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eDRUG_07.html">7. Connectivity maps enrichment analysis</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eDRUG_08.html">8. Age-drug network analysis</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eDRUG_09.html">9. Age-disease network analysis</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eDRUG_10.html">10. Funding</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eDRUG_11.html">11. Version information</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eDRUG_12.html">12. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">longevityTools_eDRUG</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eDRUG_01.html">1. Introduction</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eDRUG_02.html">2. Getting Started</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eDRUG_03.html">3. Setup of working environment</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eDRUG_04.html">4. Data downloads</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eDRUG_05.html">5. Pre-processing of CEL files</a></li>
                    

                    

                    
                    
                    
                    
                    <li class="active"><a href="../mydoc/mydoc_longevityTools_eDRUG_06.html">6. DEG analysis with `limma`</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eDRUG_07.html">7. Connectivity maps enrichment analysis</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eDRUG_08.html">8. Age-drug network analysis</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eDRUG_09.html">9. Age-disease network analysis</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eDRUG_10.html">10. Funding</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eDRUG_11.html">11. Version information</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityTools_eDRUG_12.html">12. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">longevityDrugs</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityDrugs_1.html">1. Introduction</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityDrugs_2.html">2. Getting Started</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityDrugs_3.html">3. Load database</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityDrugs_4.html">4. Query database</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityDrugs_5.html">5. Funding</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityDrugs_6.html">6. Version information</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_longevityDrugs_7.html">7. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Funding</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_funding.html">Funding</a></li>
                    

                    

                    
                    
                </ul>
                
                
                


                <!-- if you aren't using the accordion, uncomment this block:

                     <p class="external">
                         <a href="#" id="collapseAll">Collapse All</a> | <a href="#" id="expandAll">Expand All</a>
                     </p>
                 -->

    </div>
    <!-- this highlights the active parent class in the navgoco sidebar. this is critical so that the parent expands when you're viewing a page. This must appear below the sidebar code above. Otherwise, if placed inside customscripts.js, the script runs before the sidebar code runs and the class never gets inserted.-->
    <script>$("li.active").parents('li').toggleClass("active");</script>


            <!-- Content Column -->
            <div class="col-md-9">
                
                <div class="post-header">
   <h1 class="post-title-main">DEG analysis with `limma`</h1>
</div>

<div class="post-content">

   

<!-- this handles the automatic toc. use ## for subheads to auto-generate the on-page minitoc. if you use html tags, you must supply an ID for the heading element in order for it to appear in the minitoc. -->
<script>
$( document ).ready(function() {
  // Handler for .ready() called.

$('#toc').toc({ minimumHeaders: 0, listType: 'ul', showSpeed: 0, headers: 'h2,h3,h4' });

/* this offset helps account for the space taken up by the floating toolbar. */
$('#toc').on('click', 'a', function() {
  var target = $(this.getAttribute('href'))
    , scroll_target = target.offset().top

  $(window).scrollTop(scroll_target - 10);
  return false
})
  
});
</script>


<div id="toc"></div>


<!--
    
    <a target="_blank" href="https://github.com/tgirke/manuals/tree/master/mydoc/" class="btn btn-default githubEditButton" role="button"><i class="fa fa-github fa-lg"></i> Edit me</a>
    
-->
<h2 id="generate-list-of-cel-names-defining-treatment-vs-control-comparisons">Generate list of CEL names defining treatment vs. control comparisons</h2>

<p>The <code class="highlighter-rouge">sampleList</code> function extracts the sample comparisons (contrasts) from the
CMAP annotation table and stores them as a list.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">cmap</span> <span class="o">&lt;-</span> <span class="n">read.delim</span><span class="p">(</span><span class="s2">"./data/cmap_instances_02.txt"</span><span class="p">,</span> <span class="n">check.names</span><span class="o">=</span><span class="n">FALSE</span><span class="p">)</span> 
<span class="c1"># comp_list &lt;- sampleList(cmap, myby="CMP")
</span><span class="n">comp_list</span> <span class="o">&lt;-</span> <span class="n">sampleList</span><span class="p">(</span><span class="n">cmap</span><span class="p">,</span> <span class="n">myby</span><span class="o">=</span><span class="s2">"CMP_CELL"</span><span class="p">)</span></code></pre></figure>

<h2 id="load-normalized-expression-data">Load normalized expression data</h2>

<p>The following loads the MAS5 normalized expression data into a single <code class="highlighter-rouge">data.frame</code>. 
To accelerate the import, the data is read from <code class="highlighter-rouge">rds</code> files.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">chiptype_dir</span> <span class="o">&lt;-</span> <span class="n">unique</span><span class="p">(</span><span class="n">readRDS</span><span class="p">(</span><span class="s2">"./data/chiptype.rds"</span><span class="p">))</span>
<span class="n">df1</span> <span class="o">&lt;-</span> <span class="n">readRDS</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="s2">"data/"</span><span class="p">,</span> <span class="n">chiptype_dir</span><span class="p">[</span><span class="m">1</span><span class="p">],</span> <span class="s2">"/"</span><span class="p">,</span> <span class="s2">"all_mas5exprs.rds"</span><span class="p">))</span>
<span class="n">df2</span> <span class="o">&lt;-</span> <span class="n">readRDS</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="s2">"data/"</span><span class="p">,</span> <span class="n">chiptype_dir</span><span class="p">[</span><span class="m">2</span><span class="p">],</span> <span class="s2">"/"</span><span class="p">,</span> <span class="s2">"all_mas5exprs.rds"</span><span class="p">))</span>
<span class="n">df3</span> <span class="o">&lt;-</span> <span class="n">readRDS</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="s2">"data/"</span><span class="p">,</span> <span class="n">chiptype_dir</span><span class="p">[</span><span class="m">3</span><span class="p">],</span> <span class="s2">"/"</span><span class="p">,</span> <span class="s2">"all_mas5exprs.rds"</span><span class="p">))</span>
<span class="n">affyid</span> <span class="o">&lt;-</span> <span class="n">rownames</span><span class="p">(</span><span class="n">df1</span><span class="p">)[</span><span class="n">rownames</span><span class="p">(</span><span class="n">df1</span><span class="p">)</span> <span class="o">%</span><span class="k">in</span><span class="o">%</span> <span class="n">rownames</span><span class="p">(</span><span class="n">df2</span><span class="p">)];</span> <span class="n">affyid</span> <span class="o">&lt;-</span> <span class="n">affyid</span><span class="p">[</span><span class="n">affyid</span> <span class="o">%</span><span class="k">in</span><span class="o">%</span> <span class="n">rownames</span><span class="p">(</span><span class="n">df3</span><span class="p">)]</span>
<span class="n">mas5df</span> <span class="o">&lt;-</span> <span class="n">cbind</span><span class="p">(</span><span class="n">df1</span><span class="p">[</span><span class="n">affyid</span><span class="p">,],</span> <span class="n">df2</span><span class="p">[</span><span class="n">affyid</span><span class="p">,],</span> <span class="n">df3</span><span class="p">[</span><span class="n">affyid</span><span class="p">,])</span></code></pre></figure>

<h2 id="transform-probe-set-to-gene-level-data">Transform probe set to gene level data</h2>
<p>The next step generates gene level expression values. If genes are represented by several
probe sets then their mean intensities are used. This is necessary because 
the U133 chip contains many genes with duplicated probe sets. Probe sets not matching 
any gene are removed.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">myAnnot</span> <span class="o">&lt;-</span> <span class="n">readRDS</span><span class="p">(</span><span class="s2">"./results/myAnnot.rds"</span><span class="p">)</span> 
<span class="n">myAnnot</span> <span class="o">&lt;-</span> <span class="n">myAnnot</span><span class="p">[</span><span class="n">as.character</span><span class="p">(</span><span class="n">myAnnot</span><span class="p">[,</span><span class="s2">"ENTREZID"</span><span class="p">])</span> <span class="o">!=</span> <span class="s2">"NA"</span><span class="p">,]</span>
<span class="n">mas5df</span> <span class="o">&lt;-</span> <span class="n">mas5df</span><span class="p">[</span><span class="n">rownames</span><span class="p">(</span><span class="n">myAnnot</span><span class="p">),]</span>
<span class="n">idlist</span> <span class="o">&lt;-</span> <span class="n">tapply</span><span class="p">(</span><span class="n">row.names</span><span class="p">(</span><span class="n">myAnnot</span><span class="p">),</span> <span class="n">as.character</span><span class="p">(</span><span class="n">myAnnot</span><span class="o">$</span><span class="n">ENTREZID</span><span class="p">),</span> <span class="n">c</span><span class="p">)</span>
<span class="n">mas5df</span> <span class="o">&lt;-</span> <span class="n">t</span><span class="p">(</span><span class="n">sapply</span><span class="p">(</span><span class="n">names</span><span class="p">(</span><span class="n">idlist</span><span class="p">),</span> <span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="n">colMeans</span><span class="p">(</span><span class="n">mas5df</span><span class="p">[</span><span class="n">idlist</span><span class="p">[[</span><span class="n">x</span><span class="p">]],</span> <span class="p">])))</span></code></pre></figure>

<h2 id="deg-analysis-with-limma">DEG analysis with <code class="highlighter-rouge">limma</code></h2>

<p>The analysis of differentially expressed genes (DEGs) is performed with the <code class="highlighter-rouge">limma</code> package.
Genes meeting the chosen cutoff criteria are reported as DEGs (below set to FDR of 10% and 
a minimum fold change of 2). The DEG matrix is written to a file named 
<a href="http://biocluster.ucr.edu/~tgirke/projects/longevity/cmap/results/degMA.xls"><code class="highlighter-rouge">degMA.xls</code></a>.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">degList</span> <span class="o">&lt;-</span> <span class="n">runLimma</span><span class="p">(</span><span class="n">df</span><span class="o">=</span><span class="n">log2</span><span class="p">(</span><span class="n">mas5df</span><span class="p">),</span> <span class="n">comp_list</span><span class="p">,</span> <span class="n">fdr</span><span class="o">=</span><span class="m">0.10</span><span class="p">,</span> <span class="n">foldchange</span><span class="o">=</span><span class="m">1</span><span class="p">,</span> <span class="n">verbose</span><span class="o">=</span><span class="n">TRUE</span><span class="p">,</span> <span class="n">affyid</span><span class="o">=</span><span class="n">NULL</span><span class="p">)</span>
<span class="n">write.table</span><span class="p">(</span><span class="n">degList</span><span class="o">$</span><span class="n">DEG</span><span class="p">,</span> <span class="n">file</span><span class="o">=</span><span class="s2">"./results/degMA.xls"</span><span class="p">,</span> <span class="n">quote</span><span class="o">=</span><span class="n">FALSE</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"\t"</span><span class="p">,</span> <span class="n">col.names</span> <span class="o">=</span> <span class="n">NA</span><span class="p">)</span> 
<span class="n">saveRDS</span><span class="p">(</span><span class="n">degList</span><span class="o">$</span><span class="n">DEG</span><span class="p">,</span> <span class="s2">"./results/degMA.rds"</span><span class="p">)</span> <span class="c1"># saves binary matrix 
</span><span class="n">saveRDS</span><span class="p">(</span><span class="n">degList</span><span class="p">,</span> <span class="s2">"./results/degList.rds"</span><span class="p">)</span> <span class="err">#</span> <span class="n">saves</span> <span class="n">entire</span> <span class="n">degList</span></code></pre></figure>

<!-- Not neceassary anymore since intensities of genes with multiple probe sets are averaged
## Transform probe set to gene level data

The `probeset2gene` function translates a binary DEG matrix from the probe level (`row.names` are
Affy IDs) to the gene level (`row.names` are gene IDs). Genes represented by several probe
sets (rows) will be collapsed based on a chosen summary rule. The following summary rules are
supported: `summary_rule=1L` will consider a gene as a DEG for a specific treatment if at 
least one of several probe sets supports this assignment, while `summary_rule=2L` requires 
support from at least 2 probe sets.



<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">myAnnot</span> <span class="o">&lt;-</span> <span class="n">readRDS</span><span class="p">(</span><span class="s2">"./results/myAnnot.rds"</span><span class="p">)</span> 
<span class="n">degMA</span> <span class="o">&lt;-</span> <span class="n">readRDS</span><span class="p">(</span><span class="s2">"./results/degMA.rds"</span><span class="p">)</span> <span class="c1"># Faster than read.delim()
</span><span class="n">degMAgene</span> <span class="o">&lt;-</span> <span class="n">probeset2gene</span><span class="p">(</span><span class="n">degMA</span><span class="p">,</span> <span class="n">myAnnot</span><span class="p">,</span> <span class="n">geneIDtype</span><span class="o">=</span><span class="s2">"ENTREZID"</span><span class="p">,</span> <span class="n">summary_rule</span><span class="o">=</span><span class="m">1L</span><span class="p">)</span>
<span class="n">saveRDS</span><span class="p">(</span><span class="n">degMAgene</span><span class="p">,</span> <span class="s2">"./results/degMAgene.rds"</span><span class="p">)</span></code></pre></figure>

-->

<h2 id="number-of-degs-across-drug-treatments">Number of DEGs across drug treatments</h2>

<p>The following plots the number of drug treatments (y-axis) for increasing bin sizes (x-axis) 
of DEGs.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">degMAgene</span> <span class="o">&lt;-</span> <span class="n">readRDS</span><span class="p">(</span><span class="s2">"./results/degMA.rds"</span><span class="p">)</span>
<span class="n">y</span> <span class="o">&lt;-</span> <span class="n">as.numeric</span><span class="p">(</span><span class="n">colSums</span><span class="p">(</span><span class="n">degMAgene</span><span class="p">))</span>
<span class="n">interval</span> <span class="o">&lt;-</span> <span class="n">table</span><span class="p">(</span><span class="n">cut</span><span class="p">(</span><span class="n">y</span><span class="p">,</span> <span class="n">right</span><span class="o">=</span><span class="n">FALSE</span><span class="p">,</span> <span class="n">dig.lab</span><span class="o">=</span><span class="m">5</span><span class="p">,</span>  <span class="n">breaks</span><span class="o">=</span><span class="n">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span> <span class="m">5</span><span class="p">,</span> <span class="m">10</span><span class="p">,</span> <span class="m">50</span><span class="p">,</span> <span class="m">100</span><span class="p">,</span> <span class="m">200</span><span class="p">,</span> <span class="m">500</span><span class="p">,</span> <span class="m">1000</span><span class="p">,</span> <span class="m">10000</span><span class="p">)))</span>
<span class="n">df</span> <span class="o">&lt;-</span> <span class="n">data.frame</span><span class="p">(</span><span class="n">interval</span><span class="p">);</span> <span class="n">colnames</span><span class="p">(</span><span class="n">df</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">c</span><span class="p">(</span><span class="s2">"Bins"</span><span class="p">,</span> <span class="s2">"Counts"</span><span class="p">)</span>
<span class="n">ggplot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">aes</span><span class="p">(</span><span class="n">Bins</span><span class="p">,</span> <span class="n">Counts</span><span class="p">))</span> <span class="o">+</span> 
       <span class="n">geom_bar</span><span class="p">(</span><span class="n">position</span><span class="o">=</span><span class="s2">"dodge"</span><span class="p">,</span> <span class="n">stat</span><span class="o">=</span><span class="s2">"identity"</span><span class="p">,</span> <span class="n">fill</span><span class="o">=</span><span class="s2">"cornflowerblue"</span><span class="p">)</span> <span class="o">+</span> 
       <span class="n">ggtitle</span><span class="p">(</span><span class="s2">"DEG numbers by bins"</span><span class="p">)</span></code></pre></figure>

<p><img src="longevityTools_eDRUG_files/deg_distr-1.png" alt="" /></p>

<h2 id="identify-deg-overlaps-with-peters-et-al-2015">Identify DEG overlaps with Peters et al. (2015)</h2>

<p>Peters et al. (2015) reported 1,497 age-related gene expression
signatures.  The <code class="highlighter-rouge">intersectStats</code> function computes their intersects with each
of the 3,318 drug-responsive DEG sets from CMAP. The result includes the
Jaccard index as a simple similarity metric for gene sets as well as the raw
and adjusted p-values based on the hypergeometric distribution expressing how
likely it is to obtain the observed intersect sizes just by chance. The
results for the 20 top scoring drugs are given below and the full data set is
written to a file named
<a href="http://biocluster.ucr.edu/~tgirke/projects/longevity/cmap/results/degOL_PMID26490707.xls"><code class="highlighter-rouge">degOL_PMID26490707.xls</code></a>.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">PMID26490707</span> <span class="o">&lt;-</span> <span class="n">read.delim</span><span class="p">(</span><span class="s2">"./data/PMID26490707_S1.xls"</span><span class="p">,</span> <span class="n">comment</span><span class="o">=</span><span class="s2">"#"</span><span class="p">)</span>
<span class="n">myAnnot</span> <span class="o">&lt;-</span> <span class="n">readRDS</span><span class="p">(</span><span class="s2">"./results/myAnnot.rds"</span><span class="p">)</span> 
<span class="n">geneid</span> <span class="o">&lt;-</span> <span class="n">as.character</span><span class="p">(</span><span class="n">PMID26490707</span><span class="o">$</span><span class="s2">"NEW.Entrez.ID"</span><span class="p">)</span>
<span class="n">degMAgene</span> <span class="o">&lt;-</span> <span class="n">readRDS</span><span class="p">(</span><span class="s2">"./results/degMA.rds"</span><span class="p">)</span> <span class="c1"># Faster than read.delim()
</span><span class="n">degMAsub</span> <span class="o">&lt;-</span> <span class="n">degMAgene</span><span class="p">[</span><span class="n">rownames</span><span class="p">(</span><span class="n">degMAgene</span><span class="p">)</span> <span class="o">%</span><span class="k">in</span><span class="o">%</span> <span class="n">geneid</span><span class="p">,]</span>
<span class="n">degOL_PMID26490707</span> <span class="o">&lt;-</span> <span class="n">intersectStats</span><span class="p">(</span><span class="n">degMAgene</span><span class="p">,</span> <span class="n">degMAsub</span><span class="p">)</span>
<span class="n">write.table</span><span class="p">(</span><span class="n">degOL_PMID26490707</span><span class="p">,</span> <span class="n">file</span><span class="o">=</span><span class="s2">"./results/degOL_PMID26490707.xls"</span><span class="p">,</span> <span class="n">quote</span><span class="o">=</span><span class="n">FALSE</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"\t"</span><span class="p">,</span> <span class="n">col.names</span> <span class="o">=</span> <span class="n">NA</span><span class="p">)</span> 
<span class="n">sum</span><span class="p">(</span><span class="n">degOL_PMID26490707</span><span class="p">[,</span><span class="m">1</span><span class="p">]</span> <span class="o">&gt;</span> <span class="m">0</span><span class="p">)</span> <span class="err">#</span> <span class="n">Drugs</span> <span class="n">with</span> <span class="n">any</span> <span class="n">overlap</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-txt" data-lang="txt">## [1] 1137</code></pre></figure>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">degOL_PMID26490707</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">20</span><span class="p">,]</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-txt" data-lang="txt">##                                Jaccard_Index longevity_DEG cmap_DEG Intersect         Pval
## alprostadil_HL60                  0.10272537          1192      912       196 1.706881e-29
## dihydroergotamine_HL60            0.06187625          1192      404        93 3.698500e-16
## pergolide_HL60                    0.07815080          1192      767       142 2.631561e-15
## (-)-isoprenaline_HL60             0.05113636          1192      288        72 1.087773e-14
## Prestwick-983_HL60                0.05948447          1192      411        90 2.785586e-14
## nocodazole_HL60                   0.06384324          1192      491       101 4.863951e-14
## anisomycin_HL60                   0.09790419          1192     2475       327 2.312553e-11
## tetryzoline_HL60                  0.03366488          1192      159        44 4.993036e-11
## mebendazole_HL60                  0.06041924          1192      528        98 6.663275e-11
## methylergometrine_HL60            0.04436620          1192      291        63 3.825164e-10
## lycorine_HL60                     0.09093994          1192     2023       268 2.411922e-09
## podophyllotoxin_HL60              0.04175513          1192      280        59 4.108204e-09
## colchicine_HL60                   0.03181818          1192      170        42 5.960591e-09
## co-dergocrine mesilate_HL60       0.05239617          1192      455        82 1.062344e-08
## puromycin_HL60                    0.05060858          1192      448        79 5.282895e-08
## tretinoin_HL60                    0.03738318          1192      251        52 6.182212e-08
## 15-delta prostaglandin J2_HL60    0.04560698          1192      367        68 6.583181e-08
## geldanamycin_HL60                 0.03068862          1192      185        41 2.331048e-07
## bromocriptine_HL60                0.02739726          1192      158        36 6.194257e-07
## apigenin_HL60                     0.06292967          1192      818       119 1.601978e-06
##                                    adj_Pval
## alprostadil_HL60               5.936533e-26
## dihydroergotamine_HL60         1.286338e-12
## pergolide_HL60                 9.152569e-12
## (-)-isoprenaline_HL60          3.783276e-11
## Prestwick-983_HL60             9.688270e-11
## nocodazole_HL60                1.691682e-10
## anisomycin_HL60                8.043059e-08
## tetryzoline_HL60               1.736578e-07
## mebendazole_HL60               2.317487e-07
## methylergometrine_HL60         1.330392e-06
## lycorine_HL60                  8.388664e-06
## podophyllotoxin_HL60           1.428833e-05
## colchicine_HL60                2.073093e-05
## co-dergocrine mesilate_HL60    3.694832e-05
## puromycin_HL60                 1.837391e-04
## tretinoin_HL60                 2.150173e-04
## 15-delta prostaglandin J2_HL60 2.289630e-04
## geldanamycin_HL60              8.107384e-04
## bromocriptine_HL60             2.154363e-03
## apigenin_HL60                  5.571679e-03</code></pre></figure>

<h2 id="identify-deg-overlaps-with-sood-et-al-2015">Identify DEG overlaps with Sood et al. (2015)</h2>

<p>Sood et al. (2015) reported 150 age-related gene expression signatures. 
The <code class="highlighter-rouge">intersectStats</code> function computes their intersects with each of the 3,318 
drug-responsive DEG sets from CMAP. The result includes the Jaccard index as a simple 
similarity metric for gene sets as well as the raw and adjusted p-values based on the 
hypergeometric distribution expressing how likely it is to observe the observed intersect 
sizes just by chance. The results for the 20 top scoring drugs are given below and the full
data set is written to a file named <a href="http://biocluster.ucr.edu/~tgirke/projects/longevity/cmap/results/degOL_PMID26343147.xls"><code class="highlighter-rouge">degOL_PMID26343147.xls</code></a>.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">PMID26343147</span> <span class="o">&lt;-</span> <span class="n">read.delim</span><span class="p">(</span><span class="s2">"./data/PMID26343147_S1T1.xls"</span><span class="p">,</span> <span class="n">check.names</span><span class="o">=</span><span class="n">FALSE</span><span class="p">,</span> <span class="n">comment</span><span class="o">=</span><span class="s2">"#"</span><span class="p">)</span>
<span class="n">myAnnot</span> <span class="o">&lt;-</span> <span class="n">readRDS</span><span class="p">(</span><span class="s2">"./results/myAnnot.rds"</span><span class="p">)</span> 
<span class="n">geneid</span> <span class="o">&lt;-</span> <span class="n">as.character</span><span class="p">(</span><span class="n">myAnnot</span><span class="p">[</span><span class="n">rownames</span><span class="p">(</span><span class="n">myAnnot</span><span class="p">)</span> <span class="o">%</span><span class="k">in</span><span class="o">%</span> <span class="n">as.character</span><span class="p">(</span><span class="n">PMID26343147</span><span class="p">[,</span><span class="m">1</span><span class="p">]),</span> <span class="s2">"ENTREZID"</span><span class="p">])</span>
<span class="n">geneid</span> <span class="o">&lt;-</span> <span class="n">geneid</span><span class="p">[</span><span class="n">geneid</span><span class="o">!=</span><span class="s2">"NA"</span><span class="p">]</span>
<span class="n">degMA</span> <span class="o">&lt;-</span> <span class="n">readRDS</span><span class="p">(</span><span class="s2">"./results/degMA.rds"</span><span class="p">)</span> <span class="c1"># Faster then read.delim()
</span><span class="n">degMA</span> <span class="o">&lt;-</span> <span class="n">degMA</span><span class="p">[</span> <span class="p">,</span> <span class="o">!</span><span class="n">is.na</span><span class="p">(</span><span class="n">colSums</span><span class="p">(</span><span class="n">degMA</span><span class="p">))]</span> <span class="c1"># Remove columns where DEG analysis was not possible
</span><span class="n">degMAsub</span> <span class="o">&lt;-</span> <span class="n">degMA</span><span class="p">[</span><span class="n">geneid</span><span class="p">,]</span>
<span class="n">degOL_PMID26343147</span> <span class="o">&lt;-</span> <span class="n">intersectStats</span><span class="p">(</span><span class="n">degMAgene</span><span class="p">,</span> <span class="n">degMAsub</span><span class="p">)</span>
<span class="n">write.table</span><span class="p">(</span><span class="n">degOL_PMID26343147</span><span class="p">,</span> <span class="n">file</span><span class="o">=</span><span class="s2">"./results/degOL_PMID26343147.xls"</span><span class="p">,</span> <span class="n">quote</span><span class="o">=</span><span class="n">FALSE</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"\t"</span><span class="p">,</span> <span class="n">col.names</span> <span class="o">=</span> <span class="n">NA</span><span class="p">)</span> 
<span class="n">sum</span><span class="p">(</span><span class="n">degOL_PMID26343147</span><span class="p">[,</span><span class="m">1</span><span class="p">]</span> <span class="o">&gt;</span> <span class="m">0</span><span class="p">)</span> <span class="err">#</span> <span class="n">Drugs</span> <span class="n">with</span> <span class="n">any</span> <span class="n">overlap</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-txt" data-lang="txt">## [1] 223</code></pre></figure>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">degOL_PMID26343147</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">20</span><span class="p">,]</span> <span class="err">#</span> <span class="n">Top</span> <span class="m">20</span> <span class="n">scoring</span> <span class="n">drugs</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-txt" data-lang="txt">##                        Jaccard_Index longevity_DEG cmap_DEG Intersect        Pval adj_Pval
## colecalciferol_MCF7      0.017857143            55        2         1 0.008821835        1
## esculin_HL60             0.017857143            55        2         1 0.008821835        1
## flecainide_MCF7          0.017857143            55        2         1 0.008821835        1
## ribostamycin_MCF7        0.017857143            55        2         1 0.008821835        1
## withaferin A_MCF7        0.016000000            55      199         4 0.011470972        1
## dexamethasone_HL60       0.022222222            55       37         2 0.011573654        1
## lobeline_HL60            0.007987220            55     1838        15 0.011600566        1
## idoxuridine_MCF7         0.017543860            55        3         1 0.013204054        1
## metronidazole_MCF7       0.017543860            55        3         1 0.013204054        1
## sulpiride_MCF7           0.017543860            55        3         1 0.013204054        1
## ampyrone_HL60            0.008880995            55     1081        10 0.018769135        1
## chlortetracycline_HL60   0.008080808            55     1442        12 0.021597597        1
## estropipate_MCF7         0.016949153            55        5         1 0.021911500        1
## flunixin_PC3             0.016949153            55        5         1 0.021911500        1
## sulfafurazole_HL60       0.018867925            55       53         2 0.022894704        1
## (-)-MK-801_MCF7          0.016666667            55        6         1 0.026236889        1
## carbarsone_PC3           0.016666667            55        6         1 0.026236889        1
## famprofazone_HL60        0.016666667            55        6         1 0.026236889        1
## guanethidine_HL60        0.016666667            55        6         1 0.026236889        1
## cefalotin_MCF7           0.016393443            55        7         1 0.030543497        1</code></pre></figure>

<h2 id="drugs-affecting-known-longevity-genes">Drugs affecting known longevity genes</h2>

<p>The following identifies CMAP drugs affecting the expression of the IGF1 or IGF1R genes.
The final result is written to a file named <a href="http://biocluster.ucr.edu/~tgirke/projects/longevity/cmap/results/deg_IGF1.xls"><code class="highlighter-rouge">deg_IGF1.xls</code></a>.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">genesymbols</span> <span class="o">&lt;-</span> <span class="n">c</span><span class="p">(</span><span class="s2">"IGF1"</span><span class="p">,</span> <span class="s2">"IGF1R"</span><span class="p">)</span>
<span class="n">geneids</span> <span class="o">&lt;-</span> <span class="n">unique</span><span class="p">(</span><span class="n">as.character</span><span class="p">(</span><span class="n">myAnnot</span><span class="p">[</span><span class="n">myAnnot</span><span class="o">$</span><span class="n">SYMBOL</span> <span class="o">%</span><span class="k">in</span><span class="o">%</span> <span class="n">genesymbols</span><span class="p">,</span><span class="s2">"ENTREZID"</span><span class="p">]))</span>
<span class="n">names</span><span class="p">(</span><span class="n">geneids</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">unique</span><span class="p">(</span><span class="n">as.character</span><span class="p">(</span><span class="n">myAnnot</span><span class="p">[</span><span class="n">myAnnot</span><span class="o">$</span><span class="n">SYMBOL</span> <span class="o">%</span><span class="k">in</span><span class="o">%</span> <span class="n">genesymbols</span><span class="p">,</span><span class="s2">"SYMBOL"</span><span class="p">]))</span>
<span class="n">degList</span> <span class="o">&lt;-</span> <span class="n">readRDS</span><span class="p">(</span><span class="s2">"./results/degList.rds"</span><span class="p">)</span> 
<span class="n">df</span> <span class="o">&lt;-</span> <span class="n">data.frame</span><span class="p">(</span><span class="n">row.names</span><span class="o">=</span><span class="n">colnames</span><span class="p">(</span><span class="n">degList</span><span class="o">$</span><span class="n">DEG</span><span class="p">),</span> <span class="n">check.names</span><span class="o">=</span><span class="n">FALSE</span><span class="p">)</span>
<span class="n">index</span> <span class="o">&lt;-</span> <span class="n">which</span><span class="p">(</span><span class="n">colSums</span><span class="p">(</span><span class="n">degList</span><span class="o">$</span><span class="n">DEG</span><span class="p">[</span><span class="n">geneids</span><span class="p">,])</span><span class="o">&gt;=</span> <span class="m">1</span><span class="p">)</span> 
<span class="k">for</span><span class="p">(</span><span class="n">i</span> <span class="k">in</span> <span class="n">seq_along</span><span class="p">(</span><span class="n">geneids</span><span class="p">))</span> <span class="p">{</span>
    <span class="n">tmp</span> <span class="o">&lt;-</span> <span class="n">data.frame</span><span class="p">(</span><span class="n">DEG</span><span class="o">=</span><span class="n">degList</span><span class="o">$</span><span class="n">DEG</span><span class="p">[</span><span class="n">geneids</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="n">index</span><span class="p">],</span> <span class="n">logFC</span><span class="o">=</span><span class="n">degList</span><span class="o">$</span><span class="n">logFC</span><span class="p">[</span><span class="n">geneids</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="n">index</span><span class="p">],</span> <span class="n">FDR</span><span class="o">=</span><span class="n">degList</span><span class="o">$</span><span class="n">FDR</span><span class="p">[</span><span class="n">geneids</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="n">index</span><span class="p">])</span>
    <span class="n">colnames</span><span class="p">(</span><span class="n">tmp</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">paste0</span><span class="p">(</span><span class="n">names</span><span class="p">(</span><span class="n">geneids</span><span class="p">)[</span><span class="n">i</span><span class="p">],</span> <span class="s2">"_"</span><span class="p">,</span> <span class="n">colnames</span><span class="p">(</span><span class="n">tmp</span><span class="p">))</span>
    <span class="n">df</span> <span class="o">&lt;-</span> <span class="n">cbind</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">tmp</span><span class="p">[</span><span class="n">rownames</span><span class="p">(</span><span class="n">df</span><span class="p">),]</span> <span class="p">)</span>    
<span class="p">}</span>
<span class="n">df</span> <span class="o">&lt;-</span> <span class="n">df</span><span class="p">[</span><span class="n">names</span><span class="p">(</span><span class="n">index</span><span class="p">),]</span>
<span class="n">write.table</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">file</span><span class="o">=</span><span class="s2">"./results/deg_IGF1.xls"</span><span class="p">,</span> <span class="n">quote</span><span class="o">=</span><span class="n">FALSE</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"\t"</span><span class="p">,</span> <span class="n">col.names</span> <span class="o">=</span> <span class="n">NA</span><span class="p">)</span> </code></pre></figure>

<p>Now the final <code class="highlighter-rouge">data.frame</code> can be sorted by increasing mean FDR values.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">igfDF</span> <span class="o">&lt;-</span> <span class="n">read.delim</span><span class="p">(</span><span class="s2">"./results/deg_IGF1.xls"</span><span class="p">,</span> <span class="n">row.names</span><span class="o">=</span><span class="m">1</span><span class="p">)</span>
<span class="n">igfDF</span><span class="p">[</span><span class="n">order</span><span class="p">(</span><span class="n">rowMeans</span><span class="p">(</span><span class="n">igfDF</span><span class="p">[,</span><span class="n">c</span><span class="p">(</span><span class="m">3</span><span class="p">,</span><span class="m">6</span><span class="p">)])),][</span><span class="m">1</span><span class="o">:</span><span class="m">20</span><span class="p">,]</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-txt" data-lang="txt">##                                IGF1R_DEG IGF1R_logFC     IGF1R_FDR IGF1_DEG  IGF1_logFC   IGF1_FDR
## camptothecin_MCF7                      1 -1.48180690  4.867357e-07        0  0.94295576 0.07821138
## 0175029-0000_MCF7                      0 -0.62026588  1.193479e-01        1 -1.01210916 0.04295254
## cicloheximide_HL60                     1 -1.29607754  2.054384e-02        0  1.67183429 0.16908728
## emetine_HL60                           1 -1.23843896  3.085463e-03        0  1.03812083 0.26576009
## digoxigenin_PC3                        0  0.54850849  2.488776e-01        1 -1.27142976 0.04219010
## anisomycin_HL60                        1 -2.85399433  1.481347e-03        0  0.66701644 0.44025263
## vorinostat_MCF7                        1 -1.50640830  3.143781e-17        0  0.37179771 0.45437384
## irinotecan_MCF7                        1 -2.26588384  9.895790e-02        0 -1.05446828 0.41746319
## tyrphostin AG-825_MCF7                 1 -1.03570748  3.440048e-02        0  1.57208567 0.51400841
## trichostatin A_MCF7                    1 -1.27541131 6.943847e-127        0  0.07335895 0.55046789
## baclofen_HL60                          1  1.14379132  8.226627e-02        0 -1.00641253 0.53322529
## 8-azaguanine_PC3                       1 -1.04668136  7.466879e-02        0 -0.97679589 0.65221001
## verteporfin_MCF7                       1 -1.11136304  1.226251e-03        0  0.64194152 0.78463681
## piperlongumine_MCF7                    1 -1.03214332  1.088011e-02        0  0.53820854 0.78914488
## verteporfin_HL60                       1 -1.17232840  4.923492e-02        0  1.09438523 0.75358370
## cephaeline_HL60                        1 -1.13448661  1.479635e-02        0 -0.19882135 0.86619935
## lycorine_HL60                          1 -1.93807396  1.390079e-03        0 -0.23062131 0.88748758
## doxorubicin_MCF7                       1 -1.08389932  4.035039e-04        0  0.12613722 0.92452703
## anisomycin_MCF7                        0  0.06265744  8.938603e-01        1 -1.55915713 0.04355888
## 15-delta prostaglandin J2_MCF7         0  0.01396391  9.774076e-01        1  1.03626957 0.09233156</code></pre></figure>

<!-- old version


<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">genesymbols</span> <span class="o">&lt;-</span> <span class="n">c</span><span class="p">(</span><span class="s2">"IGF1"</span><span class="p">,</span> <span class="s2">"IGF1R"</span><span class="p">)</span>
<span class="n">geneids</span> <span class="o">&lt;-</span> <span class="n">unique</span><span class="p">(</span><span class="n">as.character</span><span class="p">(</span><span class="n">myAnnot</span><span class="p">[</span><span class="n">myAnnot</span><span class="o">$</span><span class="n">SYMBOL</span> <span class="o">%</span><span class="k">in</span><span class="o">%</span> <span class="n">genesymbols</span><span class="p">,</span><span class="s2">"ENTREZID"</span><span class="p">]))</span>
<span class="n">names</span><span class="p">(</span><span class="n">geneids</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">unique</span><span class="p">(</span><span class="n">as.character</span><span class="p">(</span><span class="n">myAnnot</span><span class="p">[</span><span class="n">myAnnot</span><span class="o">$</span><span class="n">SYMBOL</span> <span class="o">%</span><span class="k">in</span><span class="o">%</span> <span class="n">genesymbols</span><span class="p">,</span><span class="s2">"SYMBOL"</span><span class="p">]))</span>
<span class="n">degMAgene</span> <span class="o">&lt;-</span> <span class="n">readRDS</span><span class="p">(</span><span class="s2">"./results/degMA.rds"</span><span class="p">)</span> <span class="c1"># Faster than read.delim()
</span><span class="n">df</span> <span class="o">&lt;-</span> <span class="n">data.frame</span><span class="p">(</span><span class="n">row.names</span><span class="o">=</span><span class="n">colnames</span><span class="p">(</span><span class="n">degMAgene</span><span class="p">),</span> <span class="n">check.names</span><span class="o">=</span><span class="n">FALSE</span><span class="p">)</span>
<span class="k">for</span><span class="p">(</span><span class="n">i</span> <span class="k">in</span> <span class="n">seq_along</span><span class="p">(</span><span class="n">geneids</span><span class="p">))</span> <span class="n">df</span> <span class="o">&lt;-</span> <span class="n">cbind</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">as.numeric</span><span class="p">(</span><span class="n">degMAgene</span><span class="p">[</span><span class="n">geneids</span><span class="p">[</span><span class="n">i</span><span class="p">],]))</span>
<span class="n">colnames</span><span class="p">(</span><span class="n">df</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">names</span><span class="p">(</span><span class="n">geneids</span><span class="p">)</span>
<span class="n">df</span> <span class="o">&lt;-</span> <span class="n">df</span><span class="p">[</span><span class="n">rowSums</span><span class="p">(</span><span class="n">df</span><span class="p">)</span><span class="o">&gt;</span><span class="m">0</span><span class="p">,]</span>
<span class="n">nrow</span><span class="p">(</span><span class="n">df</span><span class="p">)</span> <span class="err">#</span> <span class="n">Number</span> <span class="n">of</span> <span class="n">drugs</span> <span class="n">affecting</span> <span class="n">at</span> <span class="n">least</span> <span class="n">one</span> <span class="n">of</span><span class="o">:</span> <span class="n">IGF1</span> <span class="n">or</span> <span class="n">IGF1R</span></code></pre></figure>



<figure class="highlight"><pre><code class="language-txt" data-lang="txt">## [1] 20</code></pre></figure>


The following computes the `limma` FDR values for the corresponding genes (here IGF1 and IGF1R) 
and drug treatments.


<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">affyids2</span> <span class="o">&lt;-</span> <span class="n">row.names</span><span class="p">(</span><span class="n">myAnnot</span><span class="p">[</span><span class="n">myAnnot</span><span class="o">$</span><span class="n">SYMBOL</span> <span class="o">%</span><span class="k">in</span><span class="o">%</span> <span class="n">genesymbols</span><span class="p">,])</span>
<span class="n">affyids</span> <span class="o">&lt;-</span> <span class="n">as.character</span><span class="p">(</span><span class="n">myAnnot</span><span class="p">[</span><span class="n">myAnnot</span><span class="o">$</span><span class="n">SYMBOL</span> <span class="o">%</span><span class="k">in</span><span class="o">%</span> <span class="n">genesymbols</span><span class="p">,</span><span class="s2">"SYMBOL"</span><span class="p">])</span>
<span class="n">names</span><span class="p">(</span><span class="n">affyids</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">affyids2</span>
<span class="n">cmap</span> <span class="o">&lt;-</span> <span class="n">read.delim</span><span class="p">(</span><span class="s2">"./data/cmap_instances_02.txt"</span><span class="p">,</span> <span class="n">check.names</span><span class="o">=</span><span class="n">FALSE</span><span class="p">)</span> 
<span class="n">comp_list</span> <span class="o">&lt;-</span> <span class="n">sampleList</span><span class="p">(</span><span class="n">cmap</span><span class="p">,</span> <span class="n">myby</span><span class="o">=</span><span class="s2">"CMP_CELL"</span><span class="p">)</span>
<span class="n">comp_list</span> <span class="o">&lt;-</span> <span class="n">comp_list</span><span class="p">[</span><span class="n">row.names</span><span class="p">(</span><span class="n">df</span><span class="p">)]</span>
<span class="n">degList</span> <span class="o">&lt;-</span> <span class="n">runLimma</span><span class="p">(</span><span class="n">df</span><span class="o">=</span><span class="n">mas5df</span><span class="p">,</span> <span class="n">comp_list</span><span class="p">,</span> <span class="n">fdr</span><span class="o">=</span><span class="m">0.10</span><span class="p">,</span> <span class="n">foldchange</span><span class="o">=</span><span class="m">1</span><span class="p">,</span> <span class="n">verbose</span><span class="o">=</span><span class="n">FALSE</span><span class="p">,</span> <span class="n">affyid</span><span class="o">=</span><span class="n">names</span><span class="p">(</span><span class="n">affyids</span><span class="p">))</span>
<span class="n">pvalDF</span> <span class="o">&lt;-</span> <span class="n">sapply</span><span class="p">(</span><span class="n">unique</span><span class="p">(</span><span class="n">affyids</span><span class="p">),</span> <span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="n">sapply</span><span class="p">(</span><span class="n">rownames</span><span class="p">(</span><span class="n">df</span><span class="p">),</span> <span class="k">function</span><span class="p">(</span><span class="n">y</span><span class="p">)</span> <span class="n">min</span><span class="p">(</span><span class="n">degList</span><span class="p">[[</span><span class="n">y</span><span class="p">]][</span><span class="n">affyids</span><span class="o">==</span><span class="n">x</span><span class="p">,</span><span class="s2">"adj.P.Val"</span><span class="p">])))</span>
<span class="n">colnames</span><span class="p">(</span><span class="n">pvalDF</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">paste0</span><span class="p">(</span><span class="n">colnames</span><span class="p">(</span><span class="n">pvalDF</span><span class="p">),</span> <span class="s2">"_FDR"</span><span class="p">)</span>
<span class="n">df</span> <span class="o">&lt;-</span> <span class="n">cbind</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">pvalDF</span><span class="p">)</span>
<span class="n">write.table</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">file</span><span class="o">=</span><span class="s2">"./results/deg_IGF1.xls"</span><span class="p">,</span> <span class="n">quote</span><span class="o">=</span><span class="n">FALSE</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"\t"</span><span class="p">,</span> <span class="n">col.names</span> <span class="o">=</span> <span class="n">NA</span><span class="p">)</span> </code></pre></figure>

-->

<h2 id="plot-structures-of-compounds">Plot structures of compounds</h2>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">library</span><span class="p">(</span><span class="n">ChemmineR</span><span class="p">)</span>
<span class="n">mypath</span> <span class="o">&lt;-</span> <span class="n">system.file</span><span class="p">(</span><span class="s2">"extdata"</span><span class="p">,</span> <span class="s2">"longevitydrugs.sdf"</span><span class="p">,</span> <span class="n">package</span><span class="o">=</span><span class="s2">"longevityTools"</span><span class="p">)</span>
<span class="n">mypath</span> <span class="o">&lt;-</span> <span class="s2">"../inst/extdata/longevitydrugs.sdf"</span>
<span class="n">sdfset</span> <span class="o">&lt;-</span> <span class="n">read.SDFset</span><span class="p">(</span><span class="n">mypath</span><span class="p">)</span>
<span class="n">data</span><span class="p">(</span><span class="n">sdfsample</span><span class="p">)</span>
<span class="n">sdfsample</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-txt" data-lang="txt">## An instance of "SDFset" with 100 molecules</code></pre></figure>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">plot</span><span class="p">(</span><span class="n">sdfsample</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">],</span> <span class="n">print</span><span class="o">=</span><span class="n">FALSE</span><span class="p">)</span></code></pre></figure>

<p><img src="longevityTools_eDRUG_files/plot_sdf-1.png" alt="" /></p>



<div class="tags">
    
</div>



</div>



<footer>
            <div class="row">
                <div class="col-lg-12 footer">
               &copy;2016 UC Riverside. All rights reserved. <br />
<p>Page last updated:</span> Tue Oct 11 17:26:57 2016<br/> Site last generated: Oct 11, 2016 <br />
<p><img src="../common_images/company_logo.png/></p>
                </div>
            </div>
</footer>




            </div><!-- /.row -->

    </div>    <!-- /.container -->

</body>


</html>

